### PR TITLE
Add missing file to compiler-rt

### PR DESCRIPTION
**Pull request by sbc100**
Also:

- Add some files to the libc blacklist.  All the these files
  transitively depend on symbols in files that are already blacklisted.
- Remove missing/redundant dependent library dependencies
- Add code to verify dependencies exist.

### CODE REVIEW

File: tools/system_libs.py [hunk 1]
```diff
@@ -605,7 +606,10 @@ def maybe_noexcept(name):
   ret = []
   has = need = None
 
+  all_names = [s[0] for s in system_libs]
+
   for shortname, suffix, create, library_symbols, deps, can_noexcept in system_libs:
+    assert all(d in all_names for d in deps)
```
Comment (kripken) [inline tools/system_libs.py]: very good idea

File: tools/shared.py [hunk 2]
```diff
@@ -1921,7 +1921,7 @@ def link_lld(args, target, opts=[], lto_level=0):
     #   cmd.append('--strip-debug')
 
     if Settings.EXPORT_ALL:
-      cmd += ['--no-gc-sections', '--export-all']
```
Comment (kripken) [inline tools/shared.py]: This looks fine if it works with the linker, but just curious, is this unrelated, or somehow connected to these changes?

Comment (sbc100) [reply tools/shared.py]: Reverted this part

**Comment by kripken** [comment]
This broke `other.test_binaryen_metadce`. I see the failure locally too.

It seems like the metadce change in this PR is what caused the breakage - why was it added?

**Pull request merged**